### PR TITLE
Allow multiple ip filters to be used

### PR DESCRIPTION
**Pull request by Merccy**
I've implemented the feature request as declared in #11

### CODE REVIEW

**Comment by pyed** [comment]
Thank you for working on this !
it'll take me sometime to review this one :)

**Comment by pyed** [comment]
make sure to run the tests before you submit a PR.

File: ipfilter.go [hunk 1]
```diff
@@ -89,22 +88,36 @@ func block(blockPage string, w *http.ResponseWriter) (int, error) {
...
+	// Create new middleware
+	newMiddleWare := func(next httpserver.Handler) httpserver.Handler {
 		return &IPFilter{
 			Next:   next,
 			Config: ifconfig,
 		}
-	}, nil
+	}
+	// Add middleware
+	cfg := httpserver.GetConfig(c.Key)
```
Comment (pyed) [inline ipfilter.go]: shouldn't this be `cfg := httpserver.GetConfig(c)` ?

File: ipfilter_test.go [hunk 2]
```diff
@@ -728,37 +823,40 @@ func TestIpfilterParse(t *testing.T) {
...
-		},
+			IsBlock:    false,
+		}, nil,
 		},
 	}
 
 	for i, test := range tests {
-		c := setup.NewTestController(test.inputIpfilterConfig)
-		actualConfig, err := ipfilterParse(c)
+		c := caddy.NewTestController(test.inputIpfilterConfig)
```
Comment (pyed) [inline ipfilter_test.go]: shouldn't this be `c := caddy.NewTestController("http", test.inputIpfilterConfig)` ?

File: ipfilter_test.go [hunk 3]
```diff
@@ -767,46 +865,136 @@ func TestIpfilterParse(t *testing.T) {
...
+			ipfilter /allowed {
+				rule allow
+				ip 192.168.1.10
+			}`, false, "212.168.23.13:12345", "/allowed", http.StatusForbidden,
+		},
+	}
+
+	for i, tc := range TestCases {
+		// Parse the text config
+		c := caddy.NewTestController(tc.inputIpfilterConfig)
```
Comment (pyed) [inline ipfilter_test.go]: same as above

**Comment by Merccy** [comment]
I ran the tests and they worked but you are right.
I'll make the changes today.

**Comment by pyed** [comment]
you are identifying which paths to check first by length, but you should check by how many slashes in the path, `/` = 0 and `/a` = 1, `/a/b` = 2, then check by the biggest value first

this test should pass
```
{
			`ipfilter / {
				rule block
				ip 192.168.1.10
			}
			ipfilter /allowed {
				rule allow
				ip 192.168.1.10
			}
			ipfilter /a/b {
				rule allow
				ip 192.168.1.13
			}`, false, "192.168.1.13:12345", "/a/b", http.StatusOK,
		},
```

**Comment by Merccy** [comment]
I don't agree.

Only the longest scope that matches is relevant.
The order is only applied when something matches.

A more specific match is always longer because a scope can't match /allowed and /a/b.
Hence sorting by length is correct.

**Comment by pyed** [comment]
That's right, my bad.
You did a great job not just implementing this feature, you refactored parts of the code, so thank you.

**Pull request closed without merging**